contrib/jinzhu/gorm/gorm.go

```diff
@@ -81,6 +81,17 @@ func WithContext(ctx context.Context, db *gorm.DB) *gorm.DB {
 	return db
 }
 
+// Context returns the context attached to a given db (via WithContext). If a
+// context has not been attached to db, returns context.Background.
+func Context(db *gorm.DB) context.Context {
```

Comment:
We want this to be a method, not a function, so
```suggestion
func (db *gorm.DB) Context() context.Context {
```

Reply:
let me know if i'm missing something, but from what i can tell we can't add a method to the `gorm.DB`

compiler error: `invalid receiver *gorm.DB (type not defined in this package)`. https://github.com/golang/go/issues/28066#issuecomment-434472394

Reply:
Oh my god! You're right! DOOOH 🤦 

Let's have `ContextFromDB` as you have suggested initially then.

Reply:
hahaha i was worried i had totally misunderstood the request! updated in 1d8689c

Reply:
and cff6b7d